### PR TITLE
server: proxy: export capture feature to a plugin 

### PR DESCRIPTION
**Pull request by kubistika**

### CODE REVIEW

File: server/proxy/modules/capture/cap_config.c [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+#include "cap_config.h"
+
+BOOL capture_plugin_init_config(captureConfig* config)
+{
+	const char* name = "PROXY_CAPTURE_TARGET";
+	char* tmp = NULL;
+	DWORD nSize = GetEnvironmentVariableA(name, NULL, 0);
+
+	if (nSize)
```
Comment (akallabeth) [inline server/proxy/modules/capture/cap_config.c:32]: integer, better use `nSize > 0`

File: server/proxy/modules/capture/cap_main.c [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+#include "pf_context.h"
+#include "cap_config.h"
+#include "cap_protocol.h"
+
+#define BUFSIZE 8092
+
+static proxyPluginsManager* g_plugins_manager = NULL;
+static captureConfig config = { 0 };
+
+static SOCKET capture_plugin_init_socket()
```
Comment (akallabeth) [inline server/proxy/modules/capture/cap_main.c:41]: in `C` this must be `(void)`

File: server/proxy/modules/capture/cap_main.c [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+static BOOL capture_plugin_session_end(proxyData* pdata)
+{
+	SOCKET socket;
+	BOOL ret;
+
+	socket = capture_plugin_get_socket(pdata);
+	if (socket == -1)
+		return FALSE;
+
+	wStream* s = capture_plugin_packet_new(SESSION_END_PDU_BASE_SIZE, MESSAGE_TYPE_SESSION_END);
```
Comment (akallabeth) [inline server/proxy/modules/capture/cap_main.c:132]: in `C` you need to declare at beginning of scope

File: server/proxy/modules/capture/cap_main.c [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+	if (!settings->SupportGraphicsPipeline)
+	{
+		WLog_ERR(TAG, "session capture is only supported for GFX clients, denying connection");
+		return FALSE;
+	}
+
+	return TRUE;
+}
+
+static BOOL capture_plugin_unload()
```
Comment (akallabeth) [inline server/proxy/modules/capture/cap_main.c:251]: must be `(void)`

**Comment by kubistika** [comment]
@akallabeth can you check please?

**Pull request merged**